ghc-lib-gen/src/Ghclibgen.hs

```diff
@@ -1467,10 +1498,8 @@ performExtraFilesSubstitutions ghcFlavor files =
 generateGhcLibParserCabal :: GhcFlavor -> [String] -> IO ()
 generateGhcLibParserCabal ghcFlavor customCppOpts = do
   (lib, _bin, parserModules, _) <- libBinParserLibModules ghcFlavor
-  -- Remove 'ghc-interal/src', 'ghc-boot-th/internal' etc. They are
-  -- dealt with later explicitly.
-  let hsSrcDirs = filter (not . isInfixOf "internal") $ ghcLibParserHsSrcDirs False ghcFlavor lib
-  let includeDirs = filter (/= "libraries/ghc-internal/include") $ ghcLibParserIncludeDirs ghcFlavor
+  let hsSrcDirs = ghcLibParserHsSrcDirs False ghcFlavor lib
+  let includeDirs = ghcLibParserIncludeDirs False ghcFlavor
```

Comment:
Same here

Reply:
`forDepends` in a context in which the source path is not going to be put to use in ghc -M. currently ignored by `ghcLibParserIncludeDirs` but available.